### ISSUE TITLE
Support Vault to inject sensitive credentials 

### ISSUE THREAD

**Issue by salsakran** [metabase/metabase#4349]
For things like encryption keys (PR #4341 ) or the keys themselves. we can use Vault backends https://www.vaultproject.io/docs/secrets/ to deal with stuff like this.

Alternatives might include jvm keystores or its ilk.

**Comment by tlrobinson** [comment]
Also Docker Secrets Management https://blog.docker.com/2017/02/docker-secrets-management/

**Comment by jeff303** [comment]
Yeah, a lot of people seem to like using JKS for this, even though it's fairly [insecure](https://www.floyd.ch/?p=1006).  For a "local file" based store, we should probably insist upon PKCS12.  A few others that I've seen "in the wild":

- [Azure Key Vault](https://azure.microsoft.com/en-us/services/key-vault/)
- [CyberArk](https://www.cyberark.com/products/secrets-manager-enterprise/)
- [Thycotic](https://www.okta.com/integrations/secret-server/) (now part of Okta)

From a UX perspective, how would this work?  I'm assuming people would somehow embed a placeholder into their database config (instead of entering the password) that would signify that field is to be loaded from a vault.  And during setup, they would presumably need to configure Metabase's access to the vault itself.
